addons/website/tools.py

```diff
@@ -41,6 +41,9 @@ def MockRequest(
             cookies=cookies,
             referrer='',
             remote_addr=remote_addr,
+            url_root=url_root,
+            url=url_root + path if url_root is not None else Mock(),
```

Comment:
1. Can't you do that after `request` is assigned the `Mock(..)` stuff in a `if`?
```
if url_root:
    request.httprequest.url = url_root + path
```

Would prevent the need to reset the `Mock()` manually.

2. Whenever you concat domain & path, use `url_join` or whatever util we use in the given odoo version (a bit like domain concat, use `expression.OR/AND`)

Reply:
Adapted :heavy_check_mark: 

>Why we need this extra line in this version forward port?

In website_menu.py", line 170, in _is_active() we do a url_parse with request.httprequest.url and wergzeug need a real thing, not a mock.